File: tests/machine/x/swift/tree_sum.swift
indirect enum Tree {
    case leaf
    case node(left: Tree, value: Int, right: Tree)
}
func sum_tree(_ t: Tree) -> Int {
    return { () in
    switch t {
    case Tree.leaf: return 0
    case let .node(left, value, right): return sum_tree(left) + value + sum_tree(right)
    }
}()
}
let t = Tree.node(left: Tree.leaf, value: 1, right: Tree.node(left: Tree.leaf, value: 2, right: Tree.leaf))
print(sum_tree(t))

File: tests/machine/x/rust/tree_sum.rs
#[derive(Debug, Clone)]
enum Tree {
        Leaf,
        Node { left: Box<Tree>, value: i32, right: Box<Tree> },
}

fn main() {
    fn sum_tree(t: &Tree) -> i32 {
        return match t {Tree::Leaf => 0, Tree::Node { left: left, value: value, right: right } => sum_tree(&left) + value + sum_tree(&right), };
    }
    let t = Tree::Node { left: Box::new(Tree::Leaf), value: 1, right: Box::new(Tree::Node { left: Box::new(Tree::Leaf), value: 2, right: Box::new(Tree::Leaf) }) };
    println!("{}", sum_tree(&t.clone()));
}
